third_party/gems/test/smoke_test.sh

```diff
@@ -3,35 +3,22 @@
 # This tests that the bundler wrapper produced by the `@gems` package is able to
 # install a gem, and provide an environment with that gem in scope.
 
+# --- begin runfiles.bash initialization v2 ---
+# Copy-pasted from the Bazel Bash runfiles library v2.
+set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
+# shellcheck disable=SC1090
+source "${RUNFILES_DIR:-/dev/null}/$f" 2>/dev/null || \
+  source "$(grep -sm1 "^$f " "${RUNFILES_MANIFEST_FILE:-/dev/null}" | cut -f2- -d' ')" 2>/dev/null || \
+  source "$0.runfiles/$f" 2>/dev/null || \
+  source "$(grep -sm1 "^$f " "$0.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+  source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+  { echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
+# --- end runfiles.bash initialization v2 ---
```

Comment:
Is it possible to change all old runfiles.bash initialization snippets to this new v2? Or is there special bazel work that must be done in tandem?

Reply:
We should be able to switch them all over. I thought that I caught them all last week, but one slipped through the cracks.

Reply:
Running `rg -l runfiles.bash` only lists five files, and they're all updated to the v2 runfiles with this PR.